DateTimeSolvedProblemsDoctav/DateTimeSolvedProblemsDoctav/Program.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            DateTime currentDate = DateTime.Now;
+            decimal totalMilliseconds = currentDate.TimeOfDay.Milliseconds + 1000 * currentDate.TimeOfDay.Seconds + 1000 * 60 * currentDate.TimeOfDay.Minutes + 1000 * 60 * 60 * currentDate.TimeOfDay.Hours;
+            Console.WriteLine($"Current date in milliseconds: {totalMilliseconds}");
+        }
+
+        //6. Write a program that calculates the date six months from the current date.
+        public static void SixMonthsFromCurrentDate()
+        {
+            DateTime currentDate = DateTime.Now;
+            DateTime nexttDate = currentDate.AddMonths(6);
+            Console.WriteLine($"{currentDate} after six months: {nexttDate}");
+        }
+
+        //7. Write a program to get the first and last second of a day.
+        public static void FirstAndLastSecondOfDay(DateTime day)
+        {
+            DateTime beginingOfTheDay = new DateTime(day.Year, day.Month, day.Day);
+            DateTime endOfTheDay = beginingOfTheDay.AddHours(24);
+            TimeSpan intervalOneMillisecond = TimeSpan.FromMilliseconds(1);
+            endOfTheDay = endOfTheDay - intervalOneMillisecond;
+            Console.WriteLine($"Beginning of the day: {beginingOfTheDay}. End of the day: {endOfTheDay}");
+        }
+
+        //8. Write a program to calculate two date difference in seconds.
+        public static void DifferenceInSeconds(DateTime date1, DateTime date2)
+        {
+            TimeSpan diferenta = date2 - date1;
+            Console.WriteLine($"The difference (seconds) between {date2} and {date1} is: {diferenta.TotalSeconds}");
+        }
+
+        //9. Given a date of birth, calculate the age of a person.
+        //Input: 10, 09, 1989 
+        //Output: 29
+        public static void CalculateAge()
+        {
+            string birth = "10, 09, 1989";
+            int day = int.Parse(birth.Substring(0, birth.IndexOf(',')));
+            birth = birth.Substring(birth.IndexOf(',') + 1);
+            int month = int.Parse(birth.Substring(0, birth.IndexOf(',')));
+            birth = birth.Substring(birth.IndexOf(',') + 1);
+            int year = int.Parse(birth);
+            //build the date in C# format
+            DateTime birthDate = new DateTime(year, month, day);
+            DateTime currentDate = DateTime.Today;
+            //calculate the age
+            int age = currentDate.Year - birthDate.Year;
+
+            if (currentDate.Month < birthDate.Month || (currentDate.Month == birthDate.Month && currentDate.Day < birthDate.Day))
+                age--;
+            Console.WriteLine($"Age = {age} years.");
+        }
+
+        //10. Write a program to get the dates 30 days before and after from the current date, and display it to the console
+        public static void DaysBeforeAfter30()
+        {
+            DateTime currentDate = DateTime.Today;
+            TimeSpan period = new TimeSpan(30, 0, 0, 0);
```

Comment:
In categoria: you deserve another sticker!

Reply:
but....daca nu ar fi timespan?

Reply:
Cred ca as apela ceva de genul:
currentDate = currentDate.Date.**AddDays(30);**